.github/workflows/ci.yml

```diff
@@ -0,0 +1,51 @@
+name: CI
+
+on:
+  push:
+    branches:
+      - master
+  pull_request:
+    branches:
+      - master
+
+jobs:
+  lint:
+    runs-on: ubuntu-latest
+    steps:
+      - name: Checkout branch
+        uses: actions/checkout@v2
+      - name: Setup default Rust toolchain
+        uses: actions-rs/toolchain@v1
+        with:
+          components: clippy,rustfmt
+      - name: Check formatting
+        run: cargo fmt --all --verbose -- --check
+      - name: Run clippy linting
+        run: cargo clippy --color=always --workspace --all-targets
+        env:
+          RUSTFLAGS:
+            "-D warnings"
+
+  build_and_test:
+    strategy:
+      matrix:
+        os:
+          - macos-latest
+          - ubuntu-latest
+        toolchain:
+          - stable
+          - beta
+          - nightly
```

Comment:
Not sure we need to test this full matrix, but I like checking macos + Linux to make sure local dev workflow doesn't break, and if this is published to crates.io it's nice to check beta + nightly toolchains as well. CI will only fail on stable+linux. Plus all this is free since it's open-source.